src/crafting_gui.cpp

```diff
@@ -241,54 +245,80 @@ const recipe *select_crafting_recipe( int &batch_size )
+                                picking = available_recipes.search( qry.substr( 2 ), recipe_subset::search_type::component );
                                 break;
 
                             case 's':
-                                current = available_recipes.search( qry.substr( 2 ), recipe_subset::search_type::skill );
+                                picking = available_recipes.search( qry.substr( 2 ), recipe_subset::search_type::skill );
                                 break;
 
                             case 'q':
-                                current = available_recipes.search( qry.substr( 2 ), recipe_subset::search_type::quality );
+                                picking = available_recipes.search( qry.substr( 2 ), recipe_subset::search_type::quality );
                                 break;
 
                             case 'Q':
-                                current = available_recipes.search( qry.substr( 2 ), recipe_subset::search_type::quality_result );
+                                picking = available_recipes.search( qry.substr( 2 ), recipe_subset::search_type::quality_result );
                                 break;
 
                             case 'm': {
                                 auto &learned = g->u.get_learned_recipes();
-                                current.clear();
                                 if( ( qry.substr( 2 ) == "yes" ) || ( qry.substr( 2 ) == "y" ) || ( qry.substr( 2 ) == "1" ) ||
                                     ( qry.substr( 2 ) == "true" ) || ( qry.substr( 2 ) == "t" ) || ( qry.substr( 2 ) == "on" ) ) {
                                     std::set_intersection( available_recipes.begin(), available_recipes.end(), learned.begin(),
-                                                           learned.end(), std::back_inserter( current ) );
+                                                           learned.end(), std::back_inserter( picking ) );
                                 } else {
                                     std::set_difference( available_recipes.begin(), available_recipes.end(), learned.begin(),
                                                          learned.end(),
-                                                         std::back_inserter( current ) );
+                                                         std::back_inserter( picking ) );
                                 }
+                                break;
+                            }
+
+                            case 'h':
+                            {
+                                picking = available_recipes.get_all();
+                                if( ( qry.substr( 2 ) == "yes" ) || ( qry.substr( 2 ) == "y" ) || ( qry.substr( 2 ) == "1" ) ||
+                                    ( qry.substr( 2 ) == "true" ) || ( qry.substr( 2 ) == "t" ) || ( qry.substr( 2 ) == "on" ) ) {
+                                    show_hidden = true;
```

Comment:
Copy-pasted.

Reply:
It's not worth making a separate function to avoid copy-pasting two lines.

Reply:
Well, in case of lines like these it totally is (even if they weren't copy-pasted, it still is). This condition would benefit from a function with a descriptive name. It also needs a tiny bit of refactoring since it _already_ is repetitive (I mean these `substr`s). By copy-pasting it you double the amount of further work (not mentioning the other drawbacks of [WET](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself#DRY_vs_WET_solutions) code).